astropy/uncertainty/core.py

```diff
@@ -0,0 +1,261 @@
+                dtype = None
+            else:
+                raise ValueError('Cannot set just dtype for a Distribution.')
+
+        result = self.distribution.view(dtype, type)
+        return Distribution(result)
+
+    def __repr__(self):
+        reprarr = repr(self.distribution)
+        if reprarr.endswith('>'):
+            firstspace = reprarr.find(' ')
+            reprarr = reprarr[firstspace+1:-1]  # :-1] removes the ending '>'
+            return '<{} {} with n_samples={}>'.format(self.__class__.__name__,
+                                                      reprarr, self.n_samples)
+        else: # numpy array-like
+            firstparen = reprarr.find('(')
+            reprarr = reprarr[firstparen:]
+            return '{}{} with n_samples={}'.format(self.__class__.__name__,
+                                                    reprarr, self.n_samples)
+            return reprarr
+
+    def __str__(self):
+        distrstr = str(self.distribution)
+        toadd = ' with n_samples={}'.format(self.n_samples)
+        return distrstr + toadd
+
+    def _repr_latex_(self):
+        superlatex = self.distribution._repr_latex_()
+        toadd = r', \; n_{{\rm samp}}={}'.format(self.n_samples)
+        return superlatex[:-1] + toadd + superlatex[-1]
+
+    @property
+    def n_samples(self):
+        """
+        The number of samples of this distribution.  A single `int`.
+        """
+        return self.dtype['samples'].shape[0]
+
+    @property
+    def pdf_mean(self):
+        """
+        The mean of this distribution.
+        """
+        return self.distribution.mean(axis=-1)
+
+    @property
+    def pdf_std(self):
+        """
+        The standard deviation of this distribution.
+        """
+        return self.distribution.std(axis=-1)
+
+    @property
+    def pdf_var(self):
+        """
+        The variance of this distribution.
+        """
+        return self.distribution.var(axis=-1)
+
+    @property
+    def pdf_median(self):
+        """
+        The median of this distribution.
+        """
+        return np.median(self.distribution, axis=-1)
+
+    @property
+    def pdf_mad(self):
+        """
+        The median absolute deviation of this distribution.
+        """
+        return np.abs(self - self.pdf_median).pdf_median
+
+    # we set this by hand because the symbolic expression (below) requires scipy
+    # _smad_scale_factor = 1 / scipy.stats.norm.ppf(0.75)
+    _smad_scale_factor = 1.48260221850560203193936104071326553821563720703125
+
+    @property
+    def pdf_smad(self):
+        """
+        The median absolute deviation of this distribution rescaled to match the
+        standard deviation for a normal distribution.
+        """
+        return self.pdf_mad * self._smad_scale_factor
+
+    def pdf_percentiles(self, perc, **kwargs):
+        """
+        Compute percentiles of this Distribution.
+
+        Parameters
+        ----------
+        perc : float or array of floats
```

Comment:
I like the `percent` option - that's a useful end-run around percentile's annoyances.  But I don't want to have our own "quantile" implementation since `numpy` now has one - would rather wait until we can use the numpy one and have *that* be `pdf_quantile`.